src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -371,7 +371,12 @@ void MergeTreePartition::serializeText(const MergeTreeData & storage, WriteBuffe
     const auto & partition_key_sample = metadata_snapshot->getPartitionKey().sample_block;
     size_t key_size = partition_key_sample.columns();
 
-    if (key_size == 0)
+    // Sometimes we can merge empty parts. In that case, the value is empty.
```

Comment:
I add some instrumentation in the `serializeText` (this function), and then run the fast test via clickhouse-test. In the end, it turns out sometimes ClickHouse is merging two empty parts. https://pastila.nl/?0042a998/918ca76ad1a0ea1bf6e30d9db583b5d3